lib/hls/hls_parser.js

```diff
@@ -1402,24 +1403,74 @@ shaka.hls.HlsParser = class {
     }
 
     let encrypted = false;
+    let hlsAes128Key;
     /** @type {!Array.<shaka.extern.DrmInfo>}*/
     const drmInfos = [];
     const keyIds = new Set();
 
-    // TODO: May still need changes to support key rotation.
+    // TODO: May still need changes to support key rotation and discontinuity.
     for (const drmTag of drmTags) {
       const method = drmTag.getRequiredAttrValue('METHOD');
-      if (method != 'NONE') {
+      if (method == 'AES-128') {
         encrypted = true;
+        this.aesEncrypted_ = true;
 
-        // We do not support AES-128 encryption with HLS yet. So, do not create
-        // StreamInfo for the playlist encrypted with AES-128.
-        // TODO: Remove the error message once we add support for AES-128.
-        if (method == 'AES-128') {
-          shaka.log.warning('Unsupported HLS Encryption', method);
-          this.aesEncrypted_ = true;
-          return null;
+        // Check if the Web Crypto API is available.
+        if (!window.crypto || !window.crypto.subtle) {
+          shaka.log.alwaysWarn('Web Crypto API is not available to decrypt ' +
+              'AES-128. (Web Crypto only exists in secure origins like https)');
+          throw new shaka.util.Error(
+              shaka.util.Error.Severity.CRITICAL,
+              shaka.util.Error.Category.MANIFEST,
+              shaka.util.Error.Code.NO_WEB_CRYPTO_API);
+        }
+
+        // HLS RFC 8216 Section 5.2:
+        // An EXT-X-KEY tag with a KEYFORMAT of "identity" that does not have an
+        // IV attribute indicates that the Media Sequence Number is to be used
+        // as the IV when decrypting a Media Segment, by putting its big-endian
+        // binary representation into a 16-octet (128-bit) buffer and padding
+        // (on the left) with zeros.
+        let firstMediaSequenceNumber = 0;
+        let iv;
+        const ivHex = drmTag.getAttributeValue('IV', '');
+        if (!ivHex) {
+          // Media Sequence Number will be used as IV.
+          firstMediaSequenceNumber =
+              shaka.hls.Utils.getFirstTagWithNameAsNumber(
+                  playlist.tags, 'EXT-X-MEDIA-SEQUENCE', 0);
+        } else {
+          // Exclude 0x at the start of string.
+          iv = shaka.util.Uint8ArrayUtils.fromHex(ivHex.substr(2));
+          if (iv.byteLength != 16) {
+            throw new shaka.util.Error(
+                shaka.util.Error.Severity.CRITICAL,
+                shaka.util.Error.Category.MANIFEST,
+                shaka.util.Error.Code.HLS_AES_128_INVALID_KEY_OR_IV_LENGTH,
+                'IV');
+          }
+        }
+
+        const keyUri = shaka.hls.Utils.constructAbsoluteUri(
+            this.masterPlaylistUri_, drmTag.getRequiredAttrValue('URI'));
+        // eslint-disable-next-line no-await-in-loop
+        const keyResponse = await this.requestManifest_(keyUri);
```

Comment:
I think this is not correct, it seems that you are trying to request a manifest, when a license request should be made. Ex: https://github.com/google/shaka-player/blob/master/lib/media/drm_engine.js#L1380

Reply:
So it would be best to create a new type, see: https://shaka-player-demo.appspot.com/docs/api/lib_net_networking_engine.js.html#line738
So in the application we can use specific filters for this.
@wjywbs what do you think?

Reply:
I hope @joeyparrish could provide some input on this.

Reply:
I think filters need to be aware of what they are playing.  License-wrapping filters, for example, shouldn't be applied to content that doesn't use licenses.

That said, it's trivial to add another request type (such as KEY or something) that is distinct from LICENSE, to make the lives of developers somewhat easier.

Reply:
I added KEY request type.